autopilot/scanner/scanner.go

```diff
@@ -164,7 +164,7 @@ func (s *scanner) fetchHosts(ctx context.Context, cutoff time.Time) chan scanJob
 
 		var exhausted bool
 		for offset := 0; !exhausted; offset += s.scanBatchSize {
-			hosts, err := s.hs.HostsForScanning(ctx, api.HostsForScanningOptions{
+			hosts, err := s.hs.Hosts(ctx, api.HostOptions{
```

Comment:
Unrelated to this PR but do we have a race here? If offset is 0 and batch size 10 but then move the offset to 10 after already scanning 5 of those hosts, we might be skipping 5 hosts that haven't been scanned yet.